nameable.rb

```diff
@@ -0,0 +1,6 @@
+# nameable class
+class Nameable
+  def correct_name
+    raise NotImplementedError
```

Comment:
- [ ] It would be very nice if you add a descriptive error message which will be shown when this exception is thrown, like the following:
```suggestion
    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```